spec/unit/rom/gateway_spec.rb

```diff
@@ -87,4 +87,38 @@ class Gateway
       expect(gateway.disconnect).to be(nil)
     end
   end
+
+  describe '.adapter' do
+    let(:gateway_class) { Class.new(ROM::Gateway) }
+
+    it 'gets/sets the adapter' do
+      expect { gateway_class.adapter :custom }
+        .to change { gateway_class.adapter }
+        .from(nil)
+        .to(:custom)
+    end
+  end
+
+  describe "#adapter" do
+    subject { ROM::CustomAdapter.new.adapter }
+
+    before do
+      ROM::CustomAdapter = Class.new(ROM::Gateway)
+    end
```

Comment:
Please define it in the test namespace which gets cleared automatically:

``` ruby
class Test::CustomGateway < ROM::Gateway
  adapter :custom
end
```

...or just use an anonymous class instead